doc/CONFIGURATION.md

```diff
@@ -141,3 +149,453 @@ node:
 # the file we will dump logs into
 log_file: ./logger.log
 ```
+
+# Datastore Configuration
+
+The `datastore` section(s) of the yaml config file is used to config key-value datastores. These key-value datastores are used throughout the entire libp2p, and ipfs stack. Every `datastore` section shares the same configuration options with each other. In fact in the code-base, they all use the same code.
+
+One thing thing to note is the section of the `node` configuration called `storage`. This is really just an aliased/renamed `datastore` section that is used for the storing all our data. By this we mean things like files, documents, movies, etc... It is for all intents and purposes our "main storage".  The one exception to this is the example configuration file earlier in the documentation which enables the `countedStore` which is a reference counted blockstore. If this is used in other `datastore` configuration sections it is simply ignored.
```

Comment:
I want to rewrite this section:

The above also applies to the `storage` section of `node`, where DAG data blocks are stored. The `storage` section also includes an additional storage type of `countedStore`, which is a reference-counted DAG block store. `countedStore` is only supported by `storage`

Reply:
good change.